### PR TITLE
[5.0] Added functionality to Validator each() to work on non-numeric indexed arrays.

### PR DESCRIPTION
**Pull request by erickubenka**
Hi guys,

regarding to #7598 i want to tackle the problem again.

When passing a string indexed array as ruleset to `each()` that one will be interpreted to work on non-numeric indexed arrays.
Otherwise the old way to build the rule will be used. 

I added a new test to show, how I think it should work. Tests are passing this time.

If you reject this solution as well, I won't send pull requests regarding to this topic again. 
Thank you for your time.

best regards,
eric

### CODE REVIEW

File: src/Illuminate/Validation/Validator.php [hunk 1]
```diff
@@ -276,9 +276,16 @@ public function each($attribute, $rules)
 
 		foreach ($data as $dataKey => $dataValue)
 		{
-			foreach ($rules as $ruleValue)
+			foreach ($rules as $ruleKey => $ruleValue)
 			{
-				$this->mergeRules("$attribute.$dataKey", $ruleValue);
+                if (!is_string($ruleKey))
```
Comment (GrahamCampbell) [inline src/Illuminate/Validation/Validator.php]: You need to indent with tabs, and you need spaces around the `!`.

File: tests/Validation/ValidationValidatorTest.php [hunk 2]
```diff
@@ -1382,14 +1382,31 @@ public function testValidateEach()
 		$data = ['foo' => [5, 10, 15]];
 
 		$v = new Validator($trans, $data, ['foo' => 'Array']);
-		$v->each('foo', ['field' => 'numeric|min:6|max:14']);
```
Comment (taylorotwell) [inline tests/Validation/ValidationValidatorTest.php]: Why did you change this test? Is this a breaking change?

Comment (erickubenka) [inline tests/Validation/ValidationValidatorTest.php]: Test would fail with changes in Validator made with this PR. 
And it is not necessary at all to specify the 'field' attribute. Without changes of this PR in Validator, you could also remove this assignment.

After this PR:

Example for normal validation rule:
'The field foo in input should be numeric'.

Example for each() validation rule without specifying the attribute:
'Each entry of field foo in input should be numeric' 

Example for each() validation rule with specified attribute: 
'Each entry of field foo in input should have numeric field bar'
